### PR TITLE
[Test Utils] Minor developer experience test improvements

### PR DESCRIPTION
**Pull request by Olshansk**
## Summary

Minor testing development environment improvements while going through the code:
- More comments
- Added `.env.dev` for sourcing
- Added an E2E test so `make test_e2e_params` is idempotent
- Added additional warning messages when running a local stress test
- Added a makefile catch all

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [ ] **Unit Tests**: `make go_develop_and_test`
- [ ] **LocalNet E2E Tests**: `make test_e2e`
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.

### CODE REVIEW

File: .env.dev [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# This file contains environment variables that are used in the development environment.
+# It is a copy of what's in the Makefile but makes local developer easier through
```
Comment (red-0ne) [inline .env.dev:2]:
```suggestion
# It is a copy of what's in the Makefile but makes local development easier through
```

File: .env.dev [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+# a simple `source .env.dev` command.
+
+POKTROLLD_HOME=./localnet/poktrolld
+# The pocket node (validator in the localnet context)
+POCKET_NODE=tcp://127.0.0.1:36657
+# TestNet RPC endpoint for validator maintained by Grove. Needs to be update if there's another "primary" testnet.
```
Comment (red-0ne) [inline .env.dev:8]:
```suggestion
# TestNet RPC endpoint for validator maintained by Grove. Needs to be updated if there's another "primary" testnet.
```

File: Makefile [hunk 3]
```diff
@@ -1,6 +1,13 @@
 .SILENT:
 
 SHELL = /bin/sh
+
+# TODO_CONSIDERATION: Should we just source `.env.dev`?
```
Comment (red-0ne) [inline Makefile:5]: This is a very good idea IMO.

File: Makefile [hunk 4]
```diff
@@ -107,6 +114,10 @@ help: ## Prints all the targets in all the Makefiles
 ### Checks ###
 ##############
 
+# TODO_DOCUMENT: All of the `check_` helpers can be installed differently depending
+# on the user's OS and enviornment.
```
Comment (red-0ne) [inline Makefile:118]:
```suggestion
# on the user's OS and environment.
```

File: e2e/tests/update_params_test.go [hunk 5]
```diff
@@ -219,9 +224,43 @@ func (s *suite) TheModuleParamShouldBeUpdated(moduleName, paramName string) {
 	s.assertExpectedModuleParamsUpdated(moduleName)
 }
 
+// AllModuleParamsShouldBeSetToTheirDefaultValues asserts that all module params are set to their default values.
```
Comment (red-0ne) [inline e2e/tests/update_params_test.go:227]:
```suggestion
// AllModuleParamsShouldBeSetToTheirDefaultValues ensures that all module params are set to their default values.
```

File: load-testing/localnet_loadtest_manifest.yaml [hunk 6]
```diff
@@ -1,39 +1,52 @@
...
   # It must be the address corresponding to the provided signing_key_name in the
   # `relay_miner_config.yaml` file.
+
+  # RelayMiner 1; http://localhost:10350/r/relayminer1/overview
   - address: pokt19a3t4yunp0dlpfjrp7qwnzwlrzd5fzs2gjaaaj
-  # The advertised URL used by the supplier when it submits a stake message on-chain.
+    # The advertised URL used by the supplier when it submits a stake message on-chain.
     exposed_url: http://relayminer1:8545
+
+  # RelayMiner 2; http://localhost:10350/r/relayminer2/overview
```
Comment (red-0ne) [inline load-testing/localnet_loadtest_manifest.yaml:26]: :+1:

File: x/proof/module/query_claim_test.go [hunk 7]
```diff
@@ -136,6 +136,10 @@ func TestClaim_List(t *testing.T) {
 	numClaimsPerSession := sharedtypes.DefaultNumBlocksPerSession
 	totalClaims := numSessions * numClaimsPerSession
 
+	// TODO_FLAKY: Ths following line in this test is flaky because test configuration
```
Comment (red-0ne) [inline x/proof/module/query_claim_test.go:139]: Would love to look into this :monocle_face:

**Review by red-0ne — CHANGES_REQUESTED** [review]
Love the `.env.dev` approach.

Left some minor changes to comments but looks good overall :+1:

File: x/proof/module/query_claim_test.go (hunk 7 above)
Comment (Olshansk) [reply x/proof/module/query_claim_test.go:139]: Assigned your name to it!

**Review by red-0ne — APPROVED** [review]
LGTM!

**Comment by red-0ne** [comment]
Pushed a couple of changes about `go_test -> test_all` in CI workflow and a reference to `go_test` in the quickstart documentation.

**Review by bryanchriswhite — COMMENTED** [review]
A bit late to the party but throwing my 2 cents in anyways.

File: Makefile [hunk 8]
```diff
@@ -382,36 +395,32 @@ test_e2e_params: test_e2e_env ## Run only the E2E suite that exercises parameter
 	go test -v ./e2e/tests/... -tags=e2e,test --features-path=update_params.feature
 
 .PHONY: test_load_relays_stress
-test_load_relays_stress: ## Run the stress test for E2E relays on non-ephemeral chains
+test_load_relays_stress: ## Run the stress test for E2E relays on a persistent (non-ephemeral) remote chains
```
[the remark this answers is not in the document]
Comment (Olshansk) [reply Makefile:398]: Done in #571

File: x/proof/module/query_claim_test.go [hunk 9]
```diff
@@ -136,6 +136,10 @@ func TestClaim_List(t *testing.T) {
 	numClaimsPerSession := sharedtypes.DefaultNumBlocksPerSession
 	totalClaims := numSessions * numClaimsPerSession
 
+	// TODO_FLAKY(@red-0ne): Ths following line in this test is flaky because test configuration
+	// and management is hard (see details above). To verify if it's functioning
+	// independently, please run:
+	// $ make itest 2 5 ./x/proof/module -- -run TestClaim_List
```
[the remark this answers is not in the document]
Comment (Olshansk) [reply x/proof/module/query_claim_test.go:142]: Will reassign to you.

File: load-testing/loadtest_manifest.yaml [hunk 10]
```diff
@@ -1,17 +1,23 @@
...
 funding_account_address: pokt14eh973xt99s7edugnyvd5d5u50d6j0hysw2vsm # address for pnf account
+
 # In non-ephemeral chains, the gateways are identified by their address.
 gateways:
   # address is the bech32 pokt address of the gateway.
   - address: pokt16sty9mjdh4u2fwgj8ptufg42cysvh6gsyx6wfp
     # The url used to send relays to the gateway on.
-    exposed_url: http://localhost:84
\ No newline at end of file
+    exposed_url: http://localhost:84
```
[the remark this answers is not in the document]
Comment (Olshansk) [reply load-testing/loadtest_manifest.yaml:23]: Pinging @okdas on discord

File: e2e/tests/update_params_test.go [hunk 11]
```diff
@@ -219,9 +224,43 @@ func (s *suite) TheModuleParamShouldBeUpdated(moduleName, paramName string) {
...
+		msgUpdateParamsValue.FieldByName("Params").Set(reflect.ValueOf(module.defaultParams))
+
+		anyMsg, err := types.NewAnyWithValue(msgUpdateParams)
+		require.NoError(s, err)
+		anyMsgs = append(anyMsgs, anyMsg)
+	}
+
+	file := s.newTempTxJSONFile(anyMsgs)
+	s.sendAuthzExecTx(s.granteeName, file.Name())
+}
```
[the remark this answers is not in the document]
Comment (Olshansk) [reply e2e/tests/update_params_test.go:259]: Done in #571.

File: e2e/tests/update_params.feature [hunk 12]
```diff
@@ -1,5 +1,10 @@
 Feature: Params Namespace
 
+  Scenario: All params are reset to their default values
+    Given the user has the pocketd binary installed
+    And an authz grant from the "gov" "module" account to the "pnf" "user" account for the "/poktroll.tokenomics.MsgUpdateParams" message exists
+    Then all module params are reset to their default values
```
[the remark this answers is not in the document]
Comment (Olshansk) [reply e2e/tests/update_params.feature:6]: Going to add a comment in #571. PTAL

**Comment by Olshansk** [comment]
@bryanchriswhite Appreciate the post-merge review. PTAL at the comments I posted in #571.

**Pull request merged**